---
title: "class14"
author: "Muhammad Tariq"
date: "2025-05-15"
output: pdf_document
---

Here we will perform a complete RNASeq analysis from counts to pathways and biological interpretation

-The data for for hands-on session comes from GEO entry: GSE37704, which is associated with the following publication:

-Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703
The authors report on differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1.

#Required Setup

```{r, message=FALSE}
library(DESeq2)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(pathview)
library(gage)
library(gageData)
```


#Data Import

```{r}
colData <- read.csv("GSE37704_metadata.csv", row.names=1)
countData <- read.csv("GSE37704_featurecounts.csv", row.names=1)
```

#Tidy counts to match 
Check the correspondence of colData rows and countData columns.
```{r}
head(countData)

counts <- countData[,-1]
```


```{r}
rownames(colData)
```
Remove the troublesome frist column so we match the metadata
```{r}
counts <- countData[,-1]
```


```{r}
all( rownames(colData) == colnames(counts))
```

#Remove zero count genes

We will have rows in'counts' for genes we can not say anything about becasue they have zero expression in particular tissue we are looking at.

```{r}
head(counts)
```

If the 'rowSum()' is zero then give gene (i.e. row) has no countdata and we should exclude these genes from further consideration

```{r}
to.keep <- rowSums(counts) != 0
cleancounts <- counts[to.keep, ]
```

>Q. How many genes do we have left?

```{r}
nrow(cleancounts)
```


#Setup for DESeq object for analysis

```{r}
dds <- DESeqDataSetFromMatrix(countData = cleancounts,
                              colData = colData,
                              design = ~condition)
```

#Run DESeq analysis

```{r}
dds = DESeq(dds)
```


#Extract the results
```{r}
res <- results(dds)
head(res)
```

#Add gene annotation
```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                     keys = rownames(res),
                     column = "SYMBOL",
                     keytype = "ENSEMBL",
                     multiVals = "first")
res$entrez <- mapIds(org.Hs.eg.db,
                     keys = rownames(res),
                     column = "ENTREZID",
                     keytype = "ENSEMBL",
                     multiVals = "first")
head(res)
```




#Save my results to CSV file

```{r}
write.csv(res, file="results.csv")
```

#Result visulatization

```{r}
library(ggplot2)
library(ggrepel)


mycols <- rep("gray", nrow(res))
mycols[res$log2FoldChange <= -2] <- "blue"
mycols[res$log2FoldChange >= +2] <- "blue"

mycols[res$padj >= 0.05 ] <= "gray"
plot(res$log2FoldChange, -log(res$padj))

abline(v=-2, col="red")
abline(v=+2, col="red")
abline(h=-log(0.05), col="red")
#mycols
plot(res$log2FoldChange, -log(res$padj), col=mycols)
```

#Pathway analysis

##

```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]

```

```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```

```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
# Examine the first 4 pathways
head(kegg.sets.hs, 4)
```

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")

```
![](hsa04110.pathview.png)

##Gene Ontology (GO)
```{r}
data(go.sets.hs)
data(go.subs.hs)

# Focus on Biological Process subset of GO
gobpsets = go.sets.hs[go.subs.hs$BP]

gobpres = gage(foldchanges, gsets=gobpsets, same.dir=TRUE)

lapply(gobpres, head)

```




##Reactance analysis online 
We need to make a little file of our significant genes that we can upload to the reactome webpage:

```{r}
sig_genes <- res[res$padj <= 0.05 & !is.na(res$padj), "symbol"]
print(paste("Total number of significant genes:", length(sig_genes)))
```
```{r}
write.table(sig_genes, file="significant_genes.txt", row.names=FALSE, col.names=FALSE, quote=FALSE)

```

```{r}
# Section 4: Reactome Analysis (Prepare gene list for online tool)

# Filter significant genes by adjusted p-value
sig_genes <- res[res$padj <= 0.05 & !is.na(res$padj), "symbol"]

# Report number of significant genes
print(paste("Total number of significant genes:", length(sig_genes)))

# Save to a plain text file for Reactome upload
write.table(sig_genes, file = "significant_genes.txt", 
            row.names = FALSE, col.names = FALSE, quote = FALSE)

```

